src/adapter/src/rbac.rs

```diff
@@ -204,15 +214,35 @@ fn generate_required_plan_attribute(plan: &Plan) -> Option<Attribute> {
         Plan::DropObjects(plan) if plan.object_type == ObjectType::Role => {
             Some(Attribute::CreateRole)
         }
-        Plan::CreateSource(_)
-        | Plan::CreateSources(_)
-        | Plan::CreateTable(_)
+        Plan::CreateSource(CreateSourcePlan { cluster_config, .. })
+        | Plan::CreateSink(CreateSinkPlan { cluster_config, .. }) => {
+            if matches!(
+                cluster_config,
+                SourceSinkClusterConfig::Linked { .. } | SourceSinkClusterConfig::Undefined
```

Comment:
I don't understand why the `Existing` variant doesn't require the create cluster attribute. (This is my first time seeing SourceSinkClusterConfig though, so I'm probably missing something obvious.)

Reply:
The `Existing` variant will create the source/sink in an existing cluster. So it's not creating a new cluster therefore it doesn't need the `CREATECLUSTER` attribute. It is creating a new object in a cluster though so it needs the `CLUSTER(U)` privilege on that specific cluster.